cicecore/cicedynB/dynamics/ice_dyn_shared.F90

```diff
@@ -749,6 +850,324 @@ end subroutine stepu
+         Tb,       & ! seabed stress factor (N/m^2)
+         uvel_init,& ! x-component of velocity (m/s), beginning of timestep
+         vvel_init,& ! y-component of velocity (m/s), beginning of timestep
+         aiu     , & ! ice fraction on [en]-grid
+         waterx  , & ! for ocean stress calculation, x (m/s)
+         watery  , & ! for ocean stress calculation, y (m/s)
+         forcex  , & ! work array: combined atm stress and ocn tilt, x
+         forcey  , & ! work array: combined atm stress and ocn tilt, y
+         massdti , & ! mass of [EN]-cell/dt (kg/m^2 s)
+         uocn    , & ! ocean current, x-direction (m/s)
+         vocn    , & ! ocean current, y-direction (m/s)
+         fm      , & ! Coriolis param. * mass in [EN]-cell (kg/s)
+         strintx , & ! divergence of internal ice stress, x (N/m^2)
+         strinty     ! divergence of internal ice stress, y (N/m^2)
+
+      real (kind=dbl_kind), dimension (nx_block,ny_block), intent(inout) :: &
+         uvel    , & ! x-component of velocity (m/s)
+         vvel        ! y-component of velocity (m/s)
+
+      real (kind=dbl_kind), dimension (nx_block,ny_block), intent(inout) :: &
+         taubx   , & ! seabed stress, x-direction (N/m^2)
+         tauby       ! seabed stress, y-direction (N/m^2)
+
+      real (kind=dbl_kind), dimension (nx_block,ny_block), intent(in) :: &
+         Cw                   ! ocean-ice neutral drag coefficient
+
+      ! local variables
+
+      integer (kind=int_kind) :: &
+         i, j, ij
+
+      real (kind=dbl_kind) :: &
+         uold, vold         , & ! old-time uvel, vvel
+         vrel               , & ! relative ice-ocean velocity
+         cca,ccb,ccc,ab2    , & ! intermediate variables
+         cc1,cc2            , & ! "
+         taux, tauy         , & ! part of ocean stress term
+         Cb                 , & ! complete seabed (basal) stress coeff
+         rhow                   !
+
+      character(len=*), parameter :: subname = '(stepuv_CD)'
+
+      !-----------------------------------------------------------------
+      ! integrate the momentum equation
+      !-----------------------------------------------------------------
+
+      call icepack_query_parameters(rhow_out=rhow)
+      call icepack_warnings_flush(nu_diag)
+      if (icepack_warnings_aborted()) call abort_ice(error_message=subname, &
+         file=__FILE__, line=__LINE__)
+
+      do ij =1, icell
+         i = indxi(ij)
+         j = indxj(ij)
+
+         uold = uvel(i,j)
+         vold = vvel(i,j)
+
+         ! (magnitude of relative ocean current)*rhow*drag*aice
+         vrel = aiu(i,j)*rhow*Cw(i,j)*sqrt((uocn(i,j) - uold)**2 + &
+                                           (vocn(i,j) - vold)**2)  ! m/s
+         ! ice/ocean stress
+         taux = vrel*waterx(i,j) ! NOTE this is not the entire
+         tauy = vrel*watery(i,j) ! ocn stress term
+
+         ccc = sqrt(uold**2 + vold**2) + u0
```

Comment:
Happy to do it, but this won't affect the cost.  Does it make the code more readable?

Reply:
yes, and there's one less local variable and copy

Reply:
There isn't really any cost in storing ccc.  This is probably all happening in scalar registers.  The hardware is doing computations and scalar register movement all the time.  Even if we get rid of ccc, there will almost certainly be a temporary created to store the value of that thing on the right hand side.  Whether that's a temporary scalar register or a temporary scalar memory location probably won't affect performance.  Also, the sqrt cost is likely to overwhelm any temporary scalar memory access.  And it may be saving the temporary to ccc concurrent with calculation.  I'm happy to do this, and do see that stepu does not define ccc, but it's probably not going to have an impact on performance.  The thing to ask is can we reduce or eliminate some sqrt calculations?